xbmc/video/VideoDatabase.cpp

```diff
@@ -1274,7 +1274,8 @@ int CVideoDatabase::GetMovieId(const std::string& strFilenameAndPath)
     if (idFile == -1)
       strSQL=PrepareSQL("select idMovie from movie join files on files.idFile=movie.idFile where files.idPath=%i",idPath);
     else
-      strSQL = PrepareSQL("select idMedia from videoversion where idFile = %i", idFile);
+      strSQL = PrepareSQL("SELECT idMedia FROM videoversion WHERE idFile = %i AND itemType = %i",
+                          idFile, VideoAssetType::VERSION);
```

Comment:
OK for this query, what about the one above (idFile == -1), not sure when it is called. Looks like an ancient code path from before 2009. PVR or addons that are not backed by a file maybe? Those didn't exist at the time though.

Reply:
Likely not, but could have been missed by xodidox. Good enough for now I suppose!